apps/src/javalab/Backpack.jsx

```diff
@@ -0,0 +1,160 @@
+import React, {Component} from 'react';
+import Radium from 'radium';
+import color from '@cdo/apps/util/color';
+import onClickOutside from 'react-onclickoutside';
+
+const placeholderFiles = [
+  'MyClass.java',
+  'MyPainter.java',
+  'NeighborhoodMain.java'
+];
+
+const styles = {
+  dropdown: {
+    position: 'absolute',
+    top: 30,
+    backgroundColor: color.darkest_gray,
+    color: color.light_gray,
+    zIndex: 20,
+    paddingTop: 5,
+    paddingBottom: 5,
+    borderRadius: 2,
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center'
+  },
+  buttonStyles: {
+    cursor: 'pointer',
+    float: 'right',
+    backgroundColor: color.light_purple,
+    marginTop: 3,
+    marginBottom: 3,
+    marginRight: 3,
+    marginLeft: 0,
+    height: 24,
+    borderRadius: 4,
+    borderWidth: 0,
+    fontSize: 13,
+    color: color.white,
+    padding: '0px 12px',
+    fontFamily: '"Gotham 5r", sans-serif',
+    lineHeight: '18px',
+    ':hover': {
+      backgroundColor: color.cyan
+    }
+  },
+  dropdownOpenButton: {
+    backgroundColor: color.cyan
+  },
+  fileListItem: {
+    display: 'flex',
+    flexDirection: 'row',
+    paddingLeft: 10,
+    paddingRight: 10,
+    paddingTop: 5,
+    paddingBottoom: 5,
+    width: '90%',
+    ':hover': {
+      backgroundColor: color.black
+    }
+  },
+  fileListLabel: {
+    marginLeft: 5
+  },
+  importButton: {
+    backgroundColor: color.orange,
+    color: color.white,
+    fontSize: 13,
+    padding: '5px 16px'
+  },
+  backpackIcon: {
+    height: 15,
+    paddingRight: 8,
+    opacity: 1
+  }
+};
+
+/**
+ * A button that drops down to a set of importable files, and closes itself if
+ * you click on the import button, or outside of the dropdown.
+ */
+export const Backpack = class Backpack extends Component {
```

Comment:
nit: i think this statement is redundant -- `export class Backpack extends Component` should work the same way